cli/cmd/install.go

```diff
@@ -973,6 +1031,82 @@ func (idopts *installIdentityOptions) genValues() (*charts.Identity, error) {
 	}, nil
 }
 
+type externalIssuerData struct {
+	trustAnchors string
+	issuerCrt    string
+	issuerKey    string
+}
+
+func loadExternalIssuerData() (*externalIssuerData, error) {
+	kubeAPI, err := k8s.NewAPI(kubeconfigPath, kubeContext, impersonate, 0)
+	if err != nil {
+		return nil, fmt.Errorf("error fetching external issuer config: %s", err)
+	}
+
+	_, err = kubeAPI.CoreV1().Namespaces().Get(controlPlaneNamespace, metav1.GetOptions{})
+	if err != nil {
+		return nil, fmt.Errorf("error fetching external issuer config: %s", err)
+	}
+
+	secret, err := kubeAPI.CoreV1().Secrets(controlPlaneNamespace).Get(k8s.IdentityIssuerSecretName, metav1.GetOptions{})
+	if secret == nil || kerrors.IsNotFound(err) {
+		return nil, fmt.Errorf("'linkerd-identity-issuer' needs to exist if --identity-external-issuer=true")
+	}
+
+	keyMissingError := "key %s containing the %s needs to exist in secret %s if --identity-external-issuer=true"
+
+	anchors, ok := secret.Data[consts.IdentityIssuerTrustAnchorsNameExternal]
+	if !ok {
+		return nil, fmt.Errorf(keyMissingError, consts.IdentityIssuerTrustAnchorsNameExternal, "trust anchors", consts.IdentityIssuerSecretName)
+	}
+
+	crt, ok := secret.Data[consts.IdentityIssuerCrtNameExternal]
+	if !ok {
+		return nil, fmt.Errorf(keyMissingError, consts.IdentityIssuerCrtNameExternal, "issuer certificate", consts.IdentityIssuerSecretName)
+	}
+
+	key, ok := secret.Data[consts.IdentityIssuerKeyNameExternal]
+	if !ok {
+		return nil, fmt.Errorf(keyMissingError, consts.IdentityIssuerKeyNameExternal, "issuer key", consts.IdentityIssuerSecretName)
+	}
+
+	return &externalIssuerData{string(anchors), string(crt), string(key)}, nil
+}
+
+func (idopts *installIdentityOptions) readExternallyManaged() (*charts.Identity, error) {
+
+	externalIssuerData, err := loadExternalIssuerData()
+	if err != nil {
+		return nil, err
+	}
+
+	creds, err := tls.ValidateAndCreateCreds(externalIssuerData.issuerCrt, externalIssuerData.issuerKey)
+
+	if err != nil {
+		log.Fatalf("Failed to read CA from %s: %s", consts.IdentityIssuerSecretName, err)
```

Comment:
For some reason, `log.Fatalf` isn't producing any output in the CLI so when there's an error the command just fails with an empty output. I would replace this (and the one below) with something like `return nil, fmt.Errorf("Failed to read CA from %s: %s", consts.IdentityIssuerSecretName, err)`